design/ceph-config-updates.md

```diff
@@ -0,0 +1,256 @@
+    daemon container has access to other daemon containers' host-persisted data. Modifying Rook's
+    behavior to only mount the individual daemon's data dir into the container as proposed here will
+    be a small security improvement on the existing behavior.
+- `--run-dir` exists for all daemons, but it is likewise more desirable to use the `/var/run/ceph`
+  path in containers. Additionally, this directory stores only unix domain sockets, and it does not
+  need to be persisted to the host. We propose to simply use the `/var/run/ceph` location in
+  containers for runtime storage of the data.
+
+### Additional configuration
+
+Additional configuration which Rook sets up initially should be done by setting values in Ceph's
+centrally-stored config. A large group of additional configurations can be configured at once via
+the Ceph command `ceph config assimilage-conf`. Care should be taken to make sure that Rook does not
+modify preexisting user-specified values.
+
+In the initial version of this implementation, Rook will set these values on every operator restart.
+This may result in user configs being overwritten but will ensure the user is not able to render
+Rook accidentally unusable. In the future, means of determining whether a user has specified a value
+or whether Rook has specified it is desired which may mean a feature addition to Ceph.
+
+Existing global configs configured here:
+- `mon allow pool delete = true`
+- `fatal signal handlers = false`
+  **TODO** is configured here, but this could be a vestigial config from Rook's old, old days
+- `log stderr prefix = "debug "` should be set for all daemons to differentiate logging from auditing
+- `debug ...` configs
+
+Removed configs:
+- `mon_max_pg_per_osd = 1000` is a dangerous setting and should be removed regardless of whether
+  this proposal is accepted
+- `log file = /dev/stderr` is set by default to keep with container standards and kept here if the
+  user needs to change this for debugging/testing
+- `mon cluster log file = /dev/stderr` for `log file` reasons above
+- `mon keyvaluedb = rocksdb` is not needed for Luminous+ clusters
+- `filestore_omap_backend = rocksdb` is not needed for Luminous+
+- `osd pg bits = 11` set (if needed) using config override for testing or play clusters
+- `osd pgp bits = 11` set (if needed) using config override for testing or play clusters
+- `osd pool default size = 1` set (if needed) using config override for testing or play clusters
+- `osd pool default min size = 1` set (if needed) using config override for testing or play clusters
+- `osd pool default pg num = 100` set (if needed) using config override for testing or play clusters
+- `osd pool default pgp num = 100` set (if needed) using config override for testing or play clusters
+- `rbd_default_features = 3` kubernetes should support Ceph's default RBD features after k8s v1.8
+
+#### Additional configs via user override
```

Comment:
@liewegas and @travisn: this section and the section immediately following contain the majority of the updates from today's huddle conversation. Please give it a good read-over and let me know if there are thoughts.

One question I have is whether the `ceph.rook.io/v2alpha1` CRD version update is appropriate or if another version should be proposed.